account_page.php

```diff
@@ -86,12 +86,19 @@
 
 html_page_top( lang_get( 'account_link' ) );
 
+$t_user_id = auth_get_current_user_id();
+
 # extracts the user information for the currently logged in user
 # and prefixes it with u_
-$t_row = user_get_row( auth_get_current_user_id() );
+$t_row = user_get_row( $t_user_id );
 
 extract( $t_row, EXTR_PREFIX_ALL, 'u' );
 
+if ( is_blank( $u_token ) ) {
+	$u_token = helper_generate_uuid();
+	user_set_field( $t_user_id, 'token', $u_token );
```

Comment:
@atrol What approach do you have in mind?

If we only disclose the token to user once, then we can use a SHA implementation.  In such case we will need to support multiple tokens, etc.

If we want to show the token to the user multiple times, then we need to encrypt / decrypt and if we do so, then the data is accessible if someone has access to config + db rather than just db in the current implementation.

If you have access to the db, then game over for MantisBT data.  However, we try to protect the password since users tend to re-use them elsewhere and we really don't need to have them beyond in the hash form.